p2p/host/resource-manager/limit_defaults.go

```diff
@@ -108,53 +111,356 @@ func (cfg *ScalingLimitConfig) AddProtocolPeerLimit(proto protocol.ID, base Base
 	}
 }
 
-type LimitConfig struct {
-	System    BaseLimit `json:",omitempty"`
-	Transient BaseLimit `json:",omitempty"`
+type LimitVal int
+
+const (
+	// DefaultLimit is the default value for resources. The exact value depends on the context, but will get values from `DefaultLimits`.
+	DefaultLimit LimitVal = 0
+	// Unlimited is the value for unlimited resources. An arbitrarily high number will also work.
+	Unlimited LimitVal = -1
```

Comment:
Purely preference. A -1 looks different than some large number `9223371036854775807` in case this is printed out. Is that large number a Max Int? (it isn't)